benchmark/Streamly/Benchmark/Data/Stream/Eliminate.hs

```diff
@@ -398,40 +426,47 @@ drainWhile = S.drainWhile (const True)
 {-# INLINE lookup #-}
 lookup :: Monad m => Int -> SerialT m Int -> m (Maybe Int)
 lookup val = S.lookup val . S.map (\x -> (x, x))
-
+#endif
 o_1_space_elimination_folds :: Int -> [Benchmark]
 o_1_space_elimination_folds value =
     [ bgroup "elimination"
         -- Basic folds
-        [ bgroup "reduce"
+        [
+#ifdef USE_PRELUDE
+            bgroup "reduce"
             [ bgroup
                   "IO"
                   [ benchIOSink value "foldl'" foldl'Reduce
                   , benchIOSink value "foldl1'" foldl1'Reduce
                   , benchIOSink value "foldlM'" foldlM'Reduce
                   ]
+
             , bgroup
                   "Identity"
                   [ benchIdentitySink value "foldl'" foldl'Reduce
                   , benchIdentitySink value "foldl1'" foldl1'Reduce
                   , benchIdentitySink value "foldlM'" foldlM'Reduce
                   ]
-            ]
-        , bgroup "build"
+            ] ,
+#endif
+         bgroup "build"
             [ bgroup "IO"
                   [ benchIOSink value "foldrMElem" (foldrMElem value)
                   ]
             , bgroup "Identity"
-                  [ benchIdentitySink value "foldrMElem" (foldrMElem value)
-                  , benchIdentitySink value "foldrToStreamLength"
-                        (S.length . runIdentity . foldrToStream)
-                  , benchPureSink value "foldrMToListLength"
+                  [ benchPureSink value "foldrMToListLength"
                         (Prelude.length . runIdentity . foldrMBuild)
+
+                  , benchIdentitySink value "foldrMElem" (foldrMElem value)
+                  , benchIdentitySink value "foldrToStreamLength"
+                        (S.fold Fold.length . runIdentity . foldrToStream)
+
```

Comment:
Why change the ordering of these benchmarks, that just adds to the diff to be reviewed.